### PR TITLE
Make Curriculum Catalog banner responsive

### PR DESCRIPTION
**Pull request by TurnerRiley**
As brought up in [this PR](https://github.com/code-dot-org/code-dot-org/pull/51249#issuecomment-1503752851), the Curriculum Catalog banner wasn't seeming as responsive when using the `HeaderBanner` component. After some investigation, this seems like the intentional behavior but I was wrapping the `HeaderBanner` component in a contianer to apply a background color (which was not contrasting well with the text that would become darker when responsive). So, to fix this, this PR adds a background image to use to maintain consistency with how `HeaderBanner` is utilized elsewhere.

In addition, this PR adds in empty alt text for design images used in the `HeaderBanner` since Meg spotted that it wasn't using alt text here.

## Demo of previous responsiveness issues
https://user-images.githubusercontent.com/9142121/231229618-53de77e5-2b7a-4a71-8e15-b7a0cb3265c6.mov

## Demo of new banner responsiveness (with comparison to show it works like it does on [studio.code.org/courses](https://studio.code.org/courses))
https://user-images.githubusercontent.com/56283563/232861998-4848d559-e268-4c48-b315-990e5dde0fb7.mp4

## Links
Jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?modal=detail&selectedIssue=ACQ-516&assignee=60d62161f65054006980bd71)

## Testing story
Local testing.

## PR Checklist:
- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by megcrenshaw** [comment]
Thanks for addressing this, Turner! It looks great, but I'm wondering if you were able to figure out what it was about the `HeaderBanner` component that wasn't working? The prop `responsiveSize` indicates it "should" work out of the box. When I look at other pages that use HeaderBanner and have subtext (https://studio.code.org/courses) and (https://studio.code.org/projects/public –– uses a component that consumes HeaderBanner), they are responsive, albeit in a slightly different way than this.

I hesitate to add more code when we have a component that should be working for us. Not sure how time sensitive this is, but how does one of the following sound?
1) Refactor HeaderBanner to accommodate for what you're trying to do. It looks like this banner is only on 6 pages total, so it theoretically shouldn't be super hard to ensure other things don't break ( ... famous last words).
2) Use HeaderBanner with additional custom styles. There's not a pattern for this already, so this might require some time figuring out the best approach (e.g. a prop that gets passed in, a separate stylesheet that overrides styles, something else?).
3) Create a follow-up ticket for this and add a comment in the HeaderBanner component that it isn't responsive in certain contexts so that others don't fall into the same scenario.

**Comment by TurnerRiley** [comment]
> Thanks for addressing this, Turner! It looks great, but I'm wondering if you were able to figure out what it was about the `HeaderBanner` component that wasn't working? The prop `responsiveSize` indicates it "should" work out of the box. When I look at other pages that use HeaderBanner and have subtext (https://studio.code.org/courses) and (https://studio.code.org/projects/public –– uses a component that consumes HeaderBanner), they are responsive, albeit in a slightly different way than this.
> 
> I hesitate to add more code when we have a component that should be working for us. Not sure how time sensitive this is, but how does one of the following sound?
> 
> 1. Refactor HeaderBanner to accommodate for what you're trying to do. It looks like this banner is only on 6 pages total, so it theoretically shouldn't be super hard to ensure other things don't break ( ... famous last words).
> 2. Use HeaderBanner with additional custom styles. There's not a pattern for this already, so this might require some time figuring out the best approach (e.g. a prop that gets passed in, a separate stylesheet that overrides styles, something else?).
> 3. Create a follow-up ticket for this and add a comment in the HeaderBanner component that it isn't responsive in certain contexts so that others don't fall into the same scenario.

While they are responsive in a slightly different way, they do exhibit the same issue that you mentioned in the initial note you made about it not being responsive (where it pushes text outside the header when it becomes too small). So I was thinking it may be good to align with Kelby's new banner designs so as to keep consistency. I will probably follow option 3 and create a ticket for looking into making HeaderBanner more responsive so that it doesn't cause the issue you mentioned and add a comment in the banner! Thank you for the feedback, and please let me know if this sounds good?

File: apps/src/templates/curriculumCatalog/CurriculumCatalog.jsx [hunk 1]
```diff
@@ -3,20 +3,29 @@ import PropTypes from 'prop-types';
...
+            {i18n.curriculumCatalogHeaderTitle()}
+          </h1>
+          <p className={style.headerSubtitle}>
+            {i18n.curriculumCatalogHeaderSubtitle()}
+          </p>
+        </div>
+        <div className={style.headerImageContainer}>
+          <img
+            className={style.headerImage}
+            src={CourseCatalogIllustration01}
```
Comment (megcrenshaw) [inline apps/src/templates/curriculumCatalog/CurriculumCatalog.jsx]: Can you add an empty alt text so that screenreaders don't read this?

**Comment by megcrenshaw** [comment]
> While they are responsive in a slightly different way, they do exhibit the same issue that you mentioned in the initial note you made about it not being responsive (where it pushes text outside the header when it becomes too small). So I was thinking it may be good to align with Kelby's new banner designs so as to keep consistency. I will probably follow option 3 and create a ticket for looking into making HeaderBanner more responsive so that it doesn't cause the issue you mentioned and add a comment in the banner! Thank you for the feedback, and please let me know if this sounds good?

I'm seeing this 
image
vs.
image

Do you see text flowing outside the header for the courses page or projects page? I don't understand why the subtext on https://studio.code.org/catalog remains whereas the subtext on https://studio.code.org/projects/public and https://studio.code.org/courses disappear when they're all using the same component.

Were you able to determine what formatting is causing the difference? It feels harder to add this code, only to refactor later with `HeaderBanner`, rather than to clean it up now.

I think if you have a day to investigate this, see where you get and if it becomes clear that it'll be a lot of extra time now to do, I'm good with the approach to add a ticket for it, adding any details you found in your investigation.

**Comment by TurnerRiley** [comment]
> > While they are responsive in a slightly different way, they do exhibit the same issue that you mentioned in the initial note you made about it not being responsive (where it pushes text outside the header when it becomes too small). So I was thinking it may be good to align with Kelby's new banner designs so as to keep consistency. I will probably follow option 3 and create a ticket for looking into making HeaderBanner more responsive so that it doesn't cause the issue you mentioned and add a comment in the banner! Thank you for the feedback, and please let me know if this sounds good?
> 
> I'm seeing this image vs. image
> 
> Do you see text flowing outside the header for the courses page or projects page? I don't understand why the subtext on https://studio.code.org/catalog remains whereas the subtext on https://studio.code.org/projects/public and https://studio.code.org/courses disappear when they're all using the same component.
> 
> Were you able to determine what formatting is causing the difference? It feels harder to add this code, only to refactor later with `HeaderBanner`, rather than to clean it up now.
> 
> I think if you have a day to investigate this, see where you get and if it becomes clear that it'll be a lot of extra time now to do, I'm good with the approach to add a ticket for it, adding any details you found in your investigation.

Oh good call! I'll investigate further and report back, thank you!

File: apps/src/templates/curriculumCatalog/CurriculumCatalog.jsx (hunk 1 above)
Comment (TurnerRiley) [reply apps/src/templates/curriculumCatalog/CurriculumCatalog.jsx]: Yes thank you for catching that!

**Comment by TurnerRiley** [comment]
> > > While they are responsive in a slightly different way, they do exhibit the same issue that you mentioned in the initial note you made about it not being responsive (where it pushes text outside the header when it becomes too small). So I was thinking it may be good to align with Kelby's new banner designs so as to keep consistency. I will probably follow option 3 and create a ticket for looking into making HeaderBanner more responsive so that it doesn't cause the issue you mentioned and add a comment in the banner! Thank you for the feedback, and please let me know if this sounds good?
> > 
> > 
> > I'm seeing this image vs. image
> > Do you see text flowing outside the header for the courses page or projects page? I don't understand why the subtext on https://studio.code.org/catalog remains whereas the subtext on https://studio.code.org/projects/public and https://studio.code.org/courses disappear when they're all using the same component.
> > Were you able to determine what formatting is causing the difference? It feels harder to add this code, only to refactor later with `HeaderBanner`, rather than to clean it up now.
> > I think if you have a day to investigate this, see where you get and if it becomes clear that it'll be a lot of extra time now to do, I'm good with the approach to add a ticket for it, adding any details you found in your investigation.
> 
> Oh good call! I'll investigate further and report back, thank you!

The reason the /catalog banner looks a bit different was because I had it nested in a container that gave it the background color (since the other banners utilize background images instead) so when the text appeared "inside" the HeaderBanner, it was actually just outside it exactly like the other pages, but the background stretched around it (making it look inside the banner).

**Comment by TurnerRiley** [comment]
So I think after some investigating I have 2 solutions that are relatively quick and doable but have their own tradeoffs and am curious your thoughts on these (or if these spark other ideas as well):

1)
Make an image that is just the background color as a rectangle (if I pass this into the HeaderBanner as the background image, this should align the styling exactly with pages like https://studio.code.org/courses, such that when the banner gets to small, it pushes the text out intentionally and changes the font color to something darker so it contrasts with the white background of the page below the banner). This will not fix what you mentioned in the previous PR (as I think this is actually the intended behavior), but it would at least unify it with other pages.
Pros:
- Way simpler PR (just a file upload and maybe 2 line changes)
Cons:
- Does not fix what you initially brought up in [the other PR](https://github.com/code-dot-org/code-dot-org/pull/51249#issuecomment-1503752851) so this would potentially require creating a new work item to adjust the behavior of HeaderBanner (however, as I mentioned above, how it works now I believe was intended and we may not want to update it alltogether)
- Would require uploading an image that is effectively the same as setting a background-color
- When the banner is shrunk to be small, the image of the computer will not show (which may not be a huge deal since it is just for design, but I'd have to check with Dani on that)

2)
Modify HeaderBanner to have a new 1-2 props such as maintainStylesWhenResponsive or something that has separate styles to make sure when text is shifted around it doesn't change color (as it normally does to contrast the white background of pages when it is pushed out of the banner).
Pros:
- This would create what we see in this PR but with fewer line changes and all within the same component
Cons:
- It would require a lot more conditionals to apply the right styling and there would be a decent amount of overlap, so reading the HeaderBanner file and trying to get the desired outcome could become pretty confusing (even with added documentation)

**Comment by megcrenshaw** [comment]
Wow, I didn't realize the subtext was getting pushed below the "header" part! Whoa. I vote for option one to keep consistent with what we already have on dashboard.

If it's clear that the product and design team is really wanting that computer, we could consider automatically maintaining the styles when responsive, not adding that option as a prop, to keep the component as simple as possible.

Two more requests (sorry):
1) Would you say that `backgroundUrl` in the HeaderBanner component is necessary to make the responsive styles work? If so, could you also make that prop required?
2) Could you add an empty alt text string to the image tag in the HeaderBanner component?

Thanks for diving deep on this!

**Comment by TurnerRiley** [comment]
> Wow, I didn't realize the subtext was getting pushed below the "header" part! Whoa. I vote for option one to keep consistent with what we already have on dashboard.
> 
> If it's clear that the product and design team is really wanting that computer, we could consider automatically maintaining the styles when responsive, not adding that option as a prop, to keep the component as simple as possible.
> 
> Two more requests (sorry):
> 
> 1. Would you say that `backgroundUrl` in the HeaderBanner component is necessary to make the responsive styles work? If so, could you also make that prop required?
> 2. Could you add an empty alt text string to the image tag in the HeaderBanner component?
> 
> Thanks for diving deep on this!

backgroundUrl is only required to the extent that the text is white so it would need something behind it to contrast so the text shows up. However, in every use of HeaderBanner (except in test files), a backgroundUrl is used, so it should be good to make it required and I'll just update the tests with them! (I'll also update the alt tag!)

File: apps/test/unit/templates/HeaderBannerTest.js [hunk 2]
```diff
@@ -21,6 +21,7 @@ function wrapperWithProps(short, subheading, description) {
         subHeadingText={subheading}
         description={description}
         short={short}
+        backgroundUrl="/shared/images/banners/courses-hero-student.jpg"
```
Comment (megcrenshaw) [inline apps/test/unit/templates/HeaderBannerTest.js:24]: Good catch!

**Review by megcrenshaw — APPROVED** [review]
Thanks for cleaning up our HeaderBanner component some! Looks good

**Pull request merged**